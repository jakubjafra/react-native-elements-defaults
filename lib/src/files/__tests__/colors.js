const testHelperUtils = require('../../testHelperUtils');

const utils = require('../../utils');
const Overwrite = utils.Overwrite;

const name = '$colors',
    __oldElement__ = require('react-native-elements/src/config/colors'),
    __element__ = require('../colors'),
    basicTheme = new Overwrite(
        name,
        Overwrite.TYPE.FILE,
        Overwrite.SCOPE.DEFAULTS,
        {
            primary: '#FFFFFF'
        }
    );

describe(name, () => {
    let OldElement, Element;

    beforeAll(() => {
        let defaultStyles = basicTheme;

        OldElement = __oldElement__.default;
        Element = __element__.compose(defaultStyles)(OldElement);
    });

    test(`config/${name} and ${name} should be different`, () => {
        expect(OldElement).not.toBe(Element);
    });

    test(`config/${name} are defined`, () => {
        expect(OldElement).toMatchSnapshot();
    });

    test(`${name} are defined`, () => {
        expect(Element).toMatchSnapshot();
    });

    test(`${name} contains overwriten property for primary color`, () => {
        expect(Element.primary).toEqual('#FFFFFF');
    })
});
