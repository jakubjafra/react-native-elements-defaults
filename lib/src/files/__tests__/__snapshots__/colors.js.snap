// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$colors $colors are defined 1`] = `
Object {
  "disabled": "#dadee0",
  "dkGreyBg": "#232323",
  "error": "#ff190c",
  "grey0": "#393e42",
  "grey1": "#43484d",
  "grey2": "#5e6977",
  "grey3": "#86939e",
  "grey4": "#bdc6cf",
  "grey5": "#e1e8ee",
  "greyOutline": "#bbb",
  "primary": "#FFFFFF",
  "primary1": "#4d86f7",
  "primary2": "#6296f9",
  "searchBg": "#303337",
  "secondary": "#8F0CE8",
  "secondary2": "#00B233",
  "secondary3": "#00FF48",
  "white": "#ffffff",
}
`;

exports[`$colors config/$colors are defined 1`] = `
Object {
  "disabled": "#dadee0",
  "dkGreyBg": "#232323",
  "error": "#ff190c",
  "grey0": "#393e42",
  "grey1": "#43484d",
  "grey2": "#5e6977",
  "grey3": "#86939e",
  "grey4": "#bdc6cf",
  "grey5": "#e1e8ee",
  "greyOutline": "#bbb",
  "primary": "#9E9E9E",
  "primary1": "#4d86f7",
  "primary2": "#6296f9",
  "searchBg": "#303337",
  "secondary": "#8F0CE8",
  "secondary2": "#00B233",
  "secondary3": "#00FF48",
  "white": "#ffffff",
}
`;
